internal/core/legacy/services/project/project.go

```diff
@@ -1282,6 +1277,78 @@ func (p *Project) convertToProjectDTO(joined bool, project *model.Project) apist
 	return projectDto
 }
 
+func (p *Project) BatchConvertProjectDTO(joined map[int64]bool, projects []model.Project) []apistructs.ProjectDTO {
+	l := logrus.WithField("func", "BatchConvertProjectDTO")
+
+	projectIDs := make([]int64, 0, len(projects))
+	for _, i := range projects {
+		projectIDs = append(projectIDs, i.ID)
+	}
+	appCount, err := p.db.GetApplicationCountByProjectIDs(projectIDs)
+	if err != nil {
+		l.WithError(err).Errorln("failed to count app")
+	}
+	appCountMap := make(map[int64]int)
+	for _, i := range appCount {
+		appCountMap[i.ID] = i.Count
+	}
+
+	memberCount, err := p.db.GetMemberCountByScopeIDs(apistructs.ProjectScope, projectIDs)
+	if err != nil {
+		l.WithError(err).Errorln("failed to count app")
+	}
+	memberCountMap := make(map[int64]int)
+	for _, i := range memberCount {
+		memberCountMap[i.ID] = i.Count
+	}
+	res := make([]apistructs.ProjectDTO, 0, len(projects))
+	for _, project := range projects {
+		var rollbackConfig map[string]int
```

Comment:
single `convertToProjectDTO` is needed, just pass in `ProjectStats`.

Reply:
done